pkg/build/node/internal/container/docker/run.go

```diff
@@ -17,77 +17,17 @@ limitations under the License.
 package docker
 
 import (
-	"sigs.k8s.io/kind/pkg/build/node/internal/container/cri"
 	"sigs.k8s.io/kind/pkg/exec"
 	"sigs.k8s.io/kind/pkg/globals"
 )
 
-// RunOpt is an option for Run
-type RunOpt func(*runOpts) *runOpts
-
-// actual options struct
-// TODO(bentheelder): these options are likely not the right or final
-type runOpts struct {
-	RunArgs       []string
-	ContainerArgs []string
-	Mounts        []cri.Mount
-	PortMappings  []cri.PortMapping
-}
-
-// WithRunArgs sets the args for docker run
-// as in the args portion of `docker run args... image containerArgs...`
-func WithRunArgs(args ...string) RunOpt {
-	return func(r *runOpts) *runOpts {
-		r.RunArgs = args
-		return r
-	}
-}
-
-// WithContainerArgs sets the args to the container
-// as in the containerArgs portion of `docker run args... image containerArgs...`
-// NOTE: this is only the args portion before the image
-func WithContainerArgs(args ...string) RunOpt {
-	return func(r *runOpts) *runOpts {
-		r.ContainerArgs = args
-		return r
-	}
-}
-
-// WithMounts sets the container mounts
-func WithMounts(mounts []cri.Mount) RunOpt {
-	return func(r *runOpts) *runOpts {
-		r.Mounts = mounts
-		return r
-	}
-}
-
-// WithPortMappings sets the container port mappings to the host
-func WithPortMappings(portMappings []cri.PortMapping) RunOpt {
-	return func(r *runOpts) *runOpts {
-		r.PortMappings = portMappings
-		return r
-	}
-}
-
 // Run creates a container with "docker run", with some error handling
-func Run(image string, opts ...RunOpt) error {
-	o := &runOpts{}
-	for _, opt := range opts {
-		o = opt(o)
-	}
-	// convert mounts to container run args
-	runArgs := o.RunArgs
-	for _, mount := range o.Mounts {
-		runArgs = append(runArgs, generateMountBindings(mount)...)
-	}
-	for _, portMapping := range o.PortMappings {
-		runArgs = append(runArgs, generatePortMappings(portMapping)...)
-	}
+func Run(image string, runArgs []string, containerArgs []string) error {
```

Comment:
can we consolidate runArgs and containerArgs?

Reply:
no, they are different things.

container args are args to the container. runArgs are options to docker run itself. they must come before and after the image respectively.

Reply:
`docker run <flags> <image> <container args>`